src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -239,7 +212,7 @@ export class Kernel implements IKernel {
         await this.startNotebook(options);
     }
     public async interrupt(): Promise<void> {
-        await Promise.all(this.eventHooks.map((h) => h('willInterrupt')));
+        await Promise.all(this.eventHooks.map((h) => h(this, 'willInterrupt')));
```

Comment:
Can we not send `this`, not a practice in general, as the calling code can easily figure this out